src/core/dynamic_batch_scheduler.h

```diff
@@ -103,6 +106,8 @@ class DynamicBatchScheduler : public Scheduler {
   std::mutex mu_;
   std::condition_variable cv_;
 
+  RateLimiter* rate_limiter_;
```

Comment:
Use shared_ptr, avoid raw pointers...

Reply:
Same as above. rate_limiter is actually held within server object already held as a raw pointer inside TritonModel.